bndtools.builder/src/org/bndtools/builder/BndtoolsBuilder.java

```diff
@@ -148,6 +148,10 @@
 						if (!Files.isSameFile(javaPath.getParent(), wsPath)) {
 							msg = "The project directory: " + javaPath + " is not a subdirectory of the bnd workspace "
 								+ wsPath;
+						} else if (!projPath.append(Project.BNDFILE)
```

Comment:
Shouldn't this be:

```java
} else if (!Files.isRegularFile(javaPath.resolve(Project.BNDFILE))) {
```

Reply:
Yes, that would probably be more efficient and more compact, thanks for the tip.

I've done this as well as added a few more different checks for weird error-causing conditions that could occur so that I can make the error message even more specific.